pkg/controller/networkpolicy/validate.go

```diff
@@ -135,6 +151,40 @@ func (v *NetworkPolicyValidator) validateCNP(curCNP, oldCNP *secv1alpha1.Cluster
 	return reason, allowed
 }
 
+// validateANP validates the admission of a Antrea NetworkPolicy resource
+func (v *NetworkPolicyValidator) validateANP(curANP, oldANP *secv1alpha1.NetworkPolicy, op admv1.Operation) (string, bool) {
+	allowed := true
+	reason := ""
+	switch op {
+	case admv1.Create:
+		// ANP "tier" must exist before referencing
+		klog.V(2).Info("Validating CREATE request for Antrea NetworkPolicy")
+		if curANP.Spec.Tier == "" || staticTierSet.Has(curANP.Spec.Tier) {
+			// Empty Tier name corresponds to default Tier
+			break
+		}
+		if ok := v.tierExists(curANP.Spec.Tier); !ok {
+			allowed = false
+			reason = fmt.Sprintf("tier %s does not exist", curANP.Spec.Tier)
+		}
+	case admv1.Update:
+		// ANP "tier" must exist before referencing
+		klog.V(2).Info("Validating UPDATE request for ANP")
+		if curANP.Spec.Tier == "" || staticTierSet.Has(curANP.Spec.Tier) {
+			// Empty Tier name corresponds to default Tier
+			break
+		}
+		if ok := v.tierExists(curANP.Spec.Tier); !ok {
+			allowed = false
+			reason = fmt.Sprintf("tier %s does not exist", curANP.Spec.Tier)
+		}
+	case admv1.Delete:
+		// Delete of ANP have no validation
+		allowed = true
+	}
+	return reason, allowed
```

Comment:
is it possible to unify that code with the one for ACNP?

e.g. with a function like this one:
```go
func validate(op admv1.Operation, tier string) (string, bool)
```

Reply:
+1, and perhaps merge Create and Update since there is only difference of a verbose log

Reply:
the only reason why i kept them separate is if at all they were to have individual validations then it make sense to keep them separate.. but we could change it later as there is no need of separate validations as of now. Done